src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+/*
+Splitting reference material can be found @
+  https://github.com/PHAREHUB/PHARE/wiki/SplitPattern
+*/
+
+#ifndef PHARE_SPLIT_2D_H
+#define PHARE_SPLIT_2D_H
+
+namespace PHARE::amr
+{
+using namespace PHARE::core;
+
+template<>
+struct ExactSplitParticleDispatcher<DimConst<2>>
+{
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals, size_t const root)
+    {
+        for (size_t y = 0; y < root; y++)
+        {
+            size_t yoff = y * root;
+            for (size_t x = 0; x < root; x++)
+            {
+                deltas[x + yoff] = {delta_vals[x], delta_vals[y]};
+            }
+        }
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>
+{
+    template<typename Weights, typename WeightVal>
+    static void set_weights(Weights& weights, WeightVal const& weight_val)
+    {
+        for (uint8_t i = 0; i < 4; i++)
+            weights[i] = weight_val;
+    }
+
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal const& delta_val)
+    {
+        deltas[0] = {-delta_val, -delta_val};
+        deltas[1] = {-delta_val, +delta_val};
+        deltas[2] = {+delta_val, -delta_val};
+        deltas[3] = {+delta_val, +delta_val};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Cross>
+{
```

Comment:
I think the `Shape` is not needed. instead of a specialization for the number of refined particles, I think we should have a `SPlitParticleDIspatcher` templeted with dim and color (among the 6 in the wiki).
Then, below, `Splitter` would also be templated with dim and color, in order to call the appropriate `SplitParticleDispatcher`, but also templated with interporder, as the coefficients depends on this value.

Reply:
The shape has been removed already.

https://github.com/Dekken/PHARE/blob/2d/src/amr/data/particles/refine/split_2d.h